tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedTestFormatTestRunner.cs

```diff
@@ -0,0 +1,304 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Tests.UnifiedTestOperations;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format
+{
+    public sealed class UnifiedTestFormatTestRunner : IDisposable
+    {
+        private EntityMap _entityMap;
+        private List<FailPoint> _failPoints = new List<FailPoint>();
+
+        [SkippableTheory]
+        [ClassData(typeof(TestCaseFactory))]
+        public void Run(JsonDrivenTestCase testCase)
+        {
+            Run(schemaVersion: testCase.Shared["schemaVersion"].AsString,
+                testSetRunOnRequirements: testCase.Shared.GetValue("runOnRequirements", null)?.AsBsonArray,
+                testRunOnRequirements: testCase.Test.GetValue("runOnRequirements", null)?.AsBsonArray,
+                entities: testCase.Shared.GetValue("createEntities", null)?.AsBsonArray,
+                initialData: testCase.Shared.GetValue("initialData", null)?.AsBsonArray,
+                test: testCase.Test);
+        }
+
+        public void Run(
+            string schemaVersion,
+            BsonArray testSetRunOnRequirements,
+            BsonArray testRunOnRequirements,
+            BsonArray entities,
+            BsonArray initialData,
+            BsonDocument test)
+        {
+            schemaVersion.Should().StartWith("1.0");
```

Comment:
Don't use `Should` for problems with the test file. Use:

    if (!schemaVersion.StartsWith("1.0"))
    {
        throw new FormatException("Schema is not 1.0.");
    }